[DOCS] `convert_model()` as a default conversion path (#17454)
openvinotoolkit/openvino: docs/Documentation/model_introduction.md

   omz_tools_downloader

Every deep learning workflow begins with obtaining a model. You can choose to prepare a custom one, use a ready-made solution and adjust it to your needs, or even download and run a pre-trained network from an online database, such as `TensorFlow Hub <https://tfhub.dev/>`__, `Hugging Face <https://huggingface.co/>`__, `Torchvision models <https://pytorch.org/hub/>`__.

:doc:`OpenVINO™ supports several model formats <Supported_Model_Formats>` and allows converting them to it's own, `openvino.runtime.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__ or `ov.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__, providing a tool dedicated to this task.

There are several options to convert a model from original framework to OpenVINO model format (``ov.Model``).

> The ``read_model()`` method converts a model from original framework to ``ov.Model``. The resulting ``ov.Model`` can be inferred in the same training environment (python script or Jupiter Notebook).

Author: > 

```suggestion
The ``read_model()`` method reads a model from a file and produces ``ov.Model``. If the file is in one of the supported original framework file formats, it is converted automatically to OpenVINO Intermediate Representation. If the file is already in an OpenVINO IR format, it is just read "as-is", without any conversion involved.
```

Model Conversion API (the ``mo.convert_model()`` method) converts a model from original framework to ``ov.Model``. The resulting ``ov.Model`` can be inferred in the same training environment (python script or Jupiter Notebook). ``convert_model()`` also has a set of parameters to :doc:`cut the model <openvino_docs_MO_DG_prepare_model_convert_model_Cutting_Model>`, :doc:`set input shapes or layout <openvino_docs_MO_DG_prepare_model_convert_model_Converting_Model>`, :doc:`add preprocessing <openvino_docs_MO_DG_Additional_Optimization_Use_Cases>`, etc.

.. image:: _static/images/model_conversion_diagram.svg
   :alt: model conversion diagram

> Another option to convert a model is to use MO command line tool. The resulting model is saved to IR (Intermediate Representation) which can be read with the ``ov.read_model()`` method, then be compiled and inferred.

Author: > Please describe somewhere that OpenVINO IR is just a serialized representation of `ov.Model`. I'm not sure that this is clearly explained somewhere.

There is already a paragraph in [Model Preparation article](https://github.com/openvinotoolkit/openvino/blob/113888cdc997b1e707d668c04dbd8e2beeced247/docs/Documentation/model_introduction.md?plain=1) , line 46.

``ov.Model`` can be serialized to IR using the ``ov.serialize()`` method. The serialized IR can be further optimized using :doc:`Post-Training Optimization tool <pot_introduction>`.

> Conversion is not required for ONNX, PaddlePaddle, TensorFlow Lite and TensorFlow models (check :doc:`TensorFlow Frontend Capabilities and Limitations <openvino_docs_MO_DG_TensorFlow_Frontend>`), as OpenVINO provides C++ and Python APIs for importing them to OpenVINO Runtime directly. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.

Author: ```suggestion
Model files (not Python objects) from ONNX, PaddlePaddle, TensorFlow and TensorFlow Lite  (check :doc:`TensorFlow Frontend Capabilities and Limitations <openvino_docs_MO_DG_TensorFlow_Frontend>`) do not require a separate step for model conversion, that is ``mo.convert_model``. OpenVINO provides C++ and Python APIs for importing the models to OpenVINO Runtime directly by just calling the ``read_model`` method. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.
```

.. image:: _static/images/model_conversion_diagram.svg
   :alt: model conversion diagram

Another option to convert a model is to use ``mo`` command-line tool. ``mo`` is a cross-platform tool that facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices in the same measure, as the ``mo.convert_model`` method,

> The resulting model is saved to IR (Intermediate Representation) which can be read with the ``ov.read_model()`` method, then be compiled and inferred.

Reviewer: only `mo` generates IR, `convert_model` returns `ov::Model` object in memory so `read_model` is not needed. May be it makes sense to be more precise in it.

> To use it, you need a pre-trained deep learning model in one of the supported formats: TensorFlow, TensorFlow Lite, PyTorch, PaddlePaddle, or ONNX. ``mo`` converts the model to the OpenVINO Intermediate Representation format (IR), which you can infer later with :doc:`OpenVINO™ Runtime <openvino_docs_OV_UG_OV_Runtime_User_Guide>`.

Reviewer: Not sure what do you mean: `To use it`. `mo` does not work with `PyTorch` format.

Another option to convert a model is to use ``mo`` command-line tool. ``mo`` is a cross-platform tool that facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices in the same measure, as the ``mo.convert_model`` method.

``mo`` requires the use of a pre-trained deep learning model in one of the supported formats: TensorFlow, TensorFlow Lite, PaddlePaddle, or ONNX. ``mo`` converts the model to the OpenVINO Intermediate Representation format (IR), which needs to be read with the ``ov.read_model()`` method. Then, you can compile and infer the ``ov.Model`` later with :doc:`OpenVINO™ Runtime <openvino_docs_OV_UG_OV_Runtime_User_Guide>`.

The figure below illustrates the typical workflow for deploying a trained deep learning model:

> .. image:: _static/images/BASIC_FLOW_MO_simplified.svg

Reviewer 2: ![image](https://github.com/openvinotoolkit/openvino/assets/29092471/596d2223-e5c8-46c5-8d9e-767ae2064bdc)

Reviewer 2: For future improvements (I believe you will apply it in a short-term): put examples for both flows near the picture to clearly see in the code how this flow is implemented. It shouldn't be exhaustive examples, but something that is correctly calls highlighted functions on the pictures.

Author: The changes have been implemented, see the commit: [21dd5fa](https://github.com/openvinotoolkit/openvino/pull/17454/commits/21dd5fa81eccd8564134ea5a65896555791a0e68)

   omz_tools_downloader

Every deep learning workflow begins with obtaining a model. You can choose to prepare a custom one, use a ready-made solution and adjust it to your needs, or even download and run a pre-trained network from an online database, such as `TensorFlow Hub <https://tfhub.dev/>`__, `Hugging Face <https://huggingface.co/>`__, `Torchvision models <https://pytorch.org/hub/>`__.

> :doc:`OpenVINO™ supports several model formats <Supported_Model_Formats>` and allows converting them to it's own, `openvino.runtime.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__ or `ov.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__, providing a tool dedicated to this task.

Reviewer 2: `openvino.runtime.Model` is the same as `ov.Model`, the latter is because we usually `import openvino.runtime as ov`. Please do not make the impression that they are two different things.

Author: > `openvino.runtime.Model` is the same as `ov.Model`, the latter is because we usually `import openvino.runtime as ov`. Please do not make the impression that they are two different things.

```suggestion
:doc:`OpenVINO™ supports several model formats <Supported_Model_Formats>` and allows converting them to it's own, `openvino.runtime.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__ (`ov.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__ ), providing a tool dedicated to this task.
```

The ``read_model()`` method reads a model from a file and produces ``ov.Model``. If the file is in one of the supported original framework file formats, it is converted automatically to OpenVINO Intermediate Representation. If the file is already in an OpenVINO IR format, it is just read "as-is", without any conversion involved.

Model Conversion API, specifically, the ``mo.convert_model()`` method converts a model from original framework to ``ov.Model``. The resulting ``ov.Model`` can be inferred in the same training environment (python script or Jupiter Notebook). In addition to model files, ``convert_model`` can take OpenVINO extension objects constructed directly in Python for easier conversion of operations that are not supported in OpenVINO. The ``convert_model()`` method also has a set of parameters to :doc:`cut the model <openvino_docs_MO_DG_prepare_model_convert_model_Cutting_Model>`, :doc:`set input shapes or layout <openvino_docs_MO_DG_prepare_model_convert_model_Converting_Model>`, :doc:`add preprocessing <openvino_docs_MO_DG_Additional_Optimization_Use_Cases>`, etc.

.. image:: _static/images/model_conversion_diagram.svg
   :alt: model conversion diagram

> Another option to convert a model is to use ``mo`` command-line tool. ``mo`` is a cross-platform tool that facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices.

Reviewer 2: `mo` does absolutely the same thing as `mo.convert_model`, it is just two APIs for the same internal thing. So `mo.convert_model` also `facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices`. Why are we describing `mo` as something that gives an extra value? This is not correct.

Author: > `mo` does absolutely the same thing as `mo.convert_model`, it is just two APIs for the same internal thing. So `mo.convert_model` also `facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices`. Why are we describing `mo` as something that gives an extra value? This is not correct.

```suggestion
Another option to convert a model is to use ``mo`` command-line tool. ``mo`` is a cross-platform tool that facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices in the same measure, as the ``mo.convert_model`` method,
```

The resulting model is saved to IR (Intermediate Representation) which can be read with the ``ov.read_model()`` method, then be compiled and inferred.

To use it, you need a pre-trained deep learning model in one of the supported formats: TensorFlow, TensorFlow Lite, PyTorch, PaddlePaddle, or ONNX. ``mo`` converts the model to the OpenVINO Intermediate Representation format (IR), which you can infer later with :doc:`OpenVINO™ Runtime <openvino_docs_OV_UG_OV_Runtime_User_Guide>`.

.. note::

   ``mo`` command-line tool does not infer models.

The figure below illustrates the typical workflow for deploying a trained deep learning model:

.. image:: _static/images/BASIC_FLOW_MO_simplified.svg

where IR is a pair of files describing the model:

* ``.xml`` - Describes the network topology.
* ``.bin`` - Contains the weights and biases binary data.

> ``ov.Model`` can be serialized to IR using the ``ov.serialize()`` method. The serialized IR can be further optimized using :doc:`Post-Training Optimization tool <pot_introduction>` that applies post-training quantization methods.

Reviewer 2: We just talked about IR, now we immediately switch back to `ov.Model`. This insertion above about `mo` in the middle between `mo.convert_model` that produces `ov.Model` and this part describing `serialize` looks not tightly connected with the context.

Author: Have rearranged it with the [a4f243d](https://github.com/openvinotoolkit/openvino/pull/17454/commits/a4f243d553e7a38ada4586623acf807e46177258 )commit.

> Model files (not Python objects) from ONNX, PaddlePaddle, TensorFlow and TensorFlow Lite  (check :doc:`TensorFlow Frontend Capabilities and Limitations <openvino_docs_MO_DG_TensorFlow_Frontend>`) do not require a separate step for model conversion, that is ``mo.convert_model``. OpenVINO provides C++ and Python APIs for importing the models to OpenVINO Runtime directly by just calling the ``read_model`` method. It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.

Reviewer 2: `It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.` -- This is more relevant to `mo.convert_model` because in framework-based code we use Python objects.

Author: > `It provides a convenient way to quickly switch from framework-based code to OpenVINO-based code in your inference application.` -- This is more relevant to `mo.convert_model` because in framework-based code we use Python objects.

```suggestion
Model files (not Python objects) from ONNX, PaddlePaddle, TensorFlow and TensorFlow Lite  (check :doc:`TensorFlow Frontend Capabilities and Limitations <openvino_docs_MO_DG_TensorFlow_Frontend>`) do not require a separate step for model conversion, that is ``mo.convert_model``. OpenVINO provides C++ and Python APIs for importing the models to OpenVINO Runtime directly by just calling the ``read_model`` method. 
```

:doc:`OpenVINO™ supports several model formats <Supported_Model_Formats>` and allows converting them to it's own, `openvino.runtime.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__ (`ov.Model <api/ie_python_api/_autosummary/openvino.runtime.Model.html>`__ ), providing a tool dedicated to this task.

There are several options to convert a model from original framework to OpenVINO model format (``ov.Model``).

The ``read_model()`` method reads a model from a file and produces ``ov.Model``. If the file is in one of the supported original framework file formats, it is converted automatically to OpenVINO Intermediate Representation. If the file is already in the OpenVINO IR format, it is read "as-is", without any conversion involved. ``ov.Model`` can be serialized to IR using the ``ov.serialize()`` method. The serialized IR can be further optimized using :doc:`Post-Training Optimization tool <pot_introduction>` that applies post-training quantization methods.

> Convert a model with ``mo`` Python API

Reviewer 2: ```suggestion
Convert a model in Python
```

.. image:: _static/images/model_conversion_diagram.svg
   :alt: model conversion diagram

Convert a model with ``mo`` command-line tool
#############################################

> Another option to convert a model is to use ``mo`` command-line tool. ``mo`` is a cross-platform tool that facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices in the same measure, as the ``mo.convert_model`` method,

Reviewer 2: Lost `,` in the end

Author: ```suggestion
Another option to convert a model is to use ``mo`` command-line tool. ``mo`` is a cross-platform tool that facilitates the transition between training and deployment environments, performs static model analysis, and adjusts deep learning models for optimal execution on end-point target devices in the same measure, as the ``mo.convert_model`` method.
```

``mo`` requires the use of a pre-trained deep learning model in one of the supported formats: TensorFlow, TensorFlow Lite, PaddlePaddle, or ONNX. ``mo`` converts the model to the OpenVINO Intermediate Representation format (IR), which needs to be read with the ``ov.read_model()`` method. Then, you can compile and infer the ``ov.Model`` later with :doc:`OpenVINO™ Runtime <openvino_docs_OV_UG_OV_Runtime_User_Guide>`.

> .. note::
> 
>    ``mo`` command-line tool does not infer models.
> 

Author: ```suggestion
```

> 

Reviewer 2: I think it is a good time to remove this funny note.

Author: Removed that part with [this](https://github.com/openvinotoolkit/openvino/pull/17454/commits/20b71555e740bd93b9a248af991effe10da8b78e) commit.
